### PR TITLE
Add additional details to UI report table.

### PR DESCRIPTION
**Pull request by wtritch**
Add additional details to the "My Reports" UI table to help users distinguish between their reports.

image

### CODE REVIEW

File: webapp/src/main/webapp/src/app/report/report.model.ts [hunk 1]
```diff
@@ -3,11 +3,17 @@
...
+import { AssessmentType } from "../shared/enum/assessment-type.enum";
+
 export class Report {
 
   public id: number;
   public label: string;
   public status: string;
+  public reportType: string;
+  public assessmentType: AssessmentType;
+  public subjectId: number;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/report/report.model.ts]: not AssessmentSubjectType?

File: webapp/src/main/webapp/src/app/report/report.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      expect(reports[0].schoolYear).toBe(1234);
+      expect(reports[0].created.getTime()).toBeLessThan(new Date().getTime());
+      expect(reports[0].assessmentType).toBeUndefined();
+      expect(reports[0].subjectId).toBe(0);
+
+      expect(reports[1].assessmentType).toBe(AssessmentType.ICA);
+      expect(reports[1].subjectId).toBe(AssessmentSubjectType.MATH);
+      done();
+    });
+  });
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/report/report.service.spec.ts]: I added a test to make sure the new model properties were being properly mapped from the API

File: webapp/src/main/webapp/src/app/report/report.model.ts [hunk 3]
```diff
@@ -3,11 +3,17 @@
...
  * or report request that can be reissued to regenerate a historic report for download again
  */
+import { AssessmentType } from "../shared/enum/assessment-type.enum";
+
 export class Report {
 
   public id: number;
   public label: string;
   public status: string;
+  public reportType: string;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/report/report.model.ts]: Should we add an enum for this too?

File: webapp/src/main/webapp/src/app/report/report.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  constructor() {
+    this.nameStudentExamReport.and.callFake((student: Student, options: ReportOptions) => {
+      return student.ssid;
+    });
+
+    this.nameGroupExamReport.and.callFake((group: Group, options: ReportOptions) => {
+      return group.name;
+    });
+
+    this.nameSchoolGradeExamReport.and.callFake((school: School, grade: Grade, options: ReportOptions) => {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/report/report.service.spec.ts]: is there an easy way in jasmine to make per-test mock responses? Seems like we always use one mock class per test file

File: webapp/src/main/webapp/src/app/report/report.model.ts (hunk 1 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/report/report.model.ts]: Ah I see -> need to represent all with 0.
We might want to make an AssessmentSubjectFilterOption enum which extends AssessmentSubjectType.
That way we don't use 0 without explanation in the code.

File: webapp/src/main/webapp/src/app/report/report.service.spec.ts (hunk 2 above)
Comment (bgsavage) [reply webapp/src/main/webapp/src/app/report/report.service.spec.ts]: I like tests :)

File: webapp/src/main/webapp/src/app/report/report.model.ts (hunk 3 above)
Comment (bgsavage) [reply webapp/src/main/webapp/src/app/report/report.model.ts]: I second

File: webapp/src/main/webapp/src/app/report/report.service.spec.ts (hunk 4 above)
Comment (wtritch) [reply webapp/src/main/webapp/src/app/report/report.service.spec.ts]: We can configure the spy in the individual tests.  I just like making a default impl of the mock methods to catch executions I'm unaware of.  
If more than one test uses/needs the same mock I'll move it into the test/ directory for re-use where it makes sense to have standard mock impls for the API methods.

File: webapp/src/main/webapp/src/app/report/report.model.ts (hunk 1 above)
Comment (wtritch) [reply webapp/src/main/webapp/src/app/report/report.model.ts]: hmm... AssessmentSubjectType is used to determine the id, here and elsewhere.  I'd rather not have 2 separate enum representations of the "same" value floating around.  I wonder if we just want to add "All" as enum value 0 to the existing AssessmentSubjectType?

File: webapp/src/main/webapp/src/app/report/report.model.ts (hunk 3 above)
Comment (wtritch) [reply webapp/src/main/webapp/src/app/report/report.model.ts]: hmm... I was following the same pattern as the "status" property.  Should they both be enums with associated display pipes, like AssessmentType?  Although, now that I look at it, the AssessmentTypePipe just returns the string value of the enum without going through translation...  Do we have a good example of enum-to-translated-display somewhere?

Comment (wtritch) [reply webapp/src/main/webapp/src/app/report/report.model.ts]: After discussion we've decided to veer away from enums towards data-driven options.

**Pull request merged**